### PR TITLE
Change country code to be more intuitive

### PR DESCRIPTION
**Pull request by BennyBlockchain**
Issue [#119](https://github.com/GaloyMoney/galoy-mobile/issues/119)

Changed the component `react-native-phone-input` to `react-native-phone-number-input` because the latter was a better design to direct the user to pick a country.

### CODE REVIEW

**Comment by BennyBlockchain** [comment]
There was also a request to add the default location to be set by the users location. I'm told the location is used in the Maps screen so it could be implemented. But, some have expressed concern asking a user for location on startup.

File: app/screens/phone-auth-screen/phone-auth.tsx [hunk 1]
```diff
@@ -103,6 +101,10 @@ const styles = EStyleSheet.create({
     textAlign: "center",
   },
 
+  textContainer: {
```
Comment (adam-stover) [inline app/screens/phone-auth-screen/phone-auth.tsx:104]: move this below `text` to address the lint error from `yarn check-code`

**Review by adam-stover — CHANGES_REQUESTED** [review]
looks good, thanks. can you fix that comment and clean up the commit history? (squash into a single commit ahead of `main`)

**Review by samerbuna — CHANGES_REQUESTED** [review]
I see some changes in this PR that are not part of it. Please rebase the changes on top of the latest in the main branch

**Comment by samerbuna** [comment]
I see big space between country flag and code:

Screen Shot 2021-12-03 at 12 30 54 PM

Also, the current component allows selecting country by typing the country code. Is there a way for this component to allow typing the country code too?

**Comment by BennyBlockchain** [comment]
> I see big space between country flag and code:
> 
> Screen Shot 2021-12-03 at 12 30 54 PM
> 
> Also, the current component allows selecting country by typing the country code. Is there a way for this component to allow typing the country code too?

The component does not allow typing in the country code. I can check on the spacing if that is available to customize.

**Comment by BennyBlockchain** [comment]
Rebased. Sorry for all the dumb pushes.

**Comment by BennyBlockchain** [comment]
@samerbuna How does this look?
Screen Shot 2021-12-03 at 4 47 50 PM

**Comment by samerbuna** [comment]
@BennyBlockchain I still see the space

**Comment by BennyBlockchain** [comment]
@samerbuna better?
Screen Shot 2021-12-04 at 12 18 12 PM
am

**Comment by samerbuna** [comment]
> @samerbuna better? 

I still see it on iOS. Have not tested Android yet.

**Comment by BennyBlockchain** [comment]
@samerbuna Just a screenshot haven't pushed changes

**Comment by samerbuna** [comment]
@BennyBlockchain I really don't like the fact that I can't select country by typing country code.

I found this library: https://github.com/MuhammedAlmaz/React-Native-INTL-Phone-Input

It would be great if we can give it a try.

**Comment by BennyBlockchain** [comment]
@samerbuna & @adam-stover 

I fixed the changes requested and added unit tests for the WelcomePhoneInput screen. Could you please review.

**Comment by daviroo** [comment]
Hi @BennyBlockchain.  Can you coordinate with @samerbuna to find out what changes are still required to get this merged?  Also the tests are failing.

**Review by samerbuna — APPROVED** [review]

**Review by daviroo — APPROVED** [review]
I've tested this locally on both android and ios and it's definitely an improvement on what we have.

**Pull request merged**